src/sandbox.rkt

```diff
@@ -35,176 +34,159 @@
 ;; If the sample contains the expected number of points, i.e., `(*num-points*) + (*reeval-pts*)`,
 ;; then the first `*num-points*` will be discarded and the rest will be used for evaluation,
 ;; otherwise the entire set is used.
-(define (get-errors test pts+exs #:seed [seed #f] #:profile [profile? #f])
-  (define output-repr (test-output-repr test))
-  (*context* (test-context test))
-  (*needed-reprs* (list output-repr (get-representation 'bool)))
-  (generate-prec-rewrites (test-conversions test))
+(define (get-errors test pcontext)
+  (setup-backend! test)
+  (unless pcontext
+    (error 'get-errors "cannnot run without a pcontext"))
 
-  (when seed (set-seed! seed))
-  (random) ;; Child process uses deterministic but different seed from evaluator
-
-  (define-values (joint-pcontext train-pcontext test-pcontext)
-    (compute-pcontexts pts+exs (*context*)))
-
-  (define processed-pcontext (preprocess-pcontext test-pcontext (*herbie-preprocess*) (*context*)))
-  (define errs (errors (test-input test) processed-pcontext (*context*)))
+  (define-values (_ test-pcontext) (partition-pcontext pcontext (*context*)))
+  (define errs (errors (test-input test) test-pcontext (*context*)))
 
   (for/list ([(pt _) (in-pcontext test-pcontext)] [err (in-list errs)])
     (list pt (format-bits (ulps->bits err)))))
 
-;; Given a test and a sample of points, computes the local error at every node in the expression
-;; returning a tree of errors that mirrors the structure of the expression.
+;; Given a test and a sample of points, the ground truth of each point
 ;; If the sample contains the expected number of points, i.e., `(*num-points*) + (*reeval-pts*)`,
 ;; then the first `*num-points*` will be discarded and the rest will be used for evaluation,
 ;; otherwise the entire set is used.
-(define (get-local-error test pts+exs #:seed [seed #f] #:profile [profile? #f])
-  (define output-repr (test-output-repr test))
-  (*context* (test-context test))
-  (*needed-reprs* (list output-repr (get-representation 'bool)))
-  (generate-prec-rewrites (test-conversions test))
+(define (get-exacts test pcontext)
+  (setup-backend! test)
+  (unless pcontext
+    (error 'get-exacts "cannnot run without a pcontext"))
 
-  (when seed (set-seed! seed))
-  (random) ;; Child process uses deterministic but different seed from evaluator
+  (define repr (test-output-repr test))
+  (define-values (train-pcontext test-pcontext) (partition-pcontext pcontext (*context*)))
+  (define-values (pts _) (pcontext->lists test-pcontext))
 
-  (define-values (joint-pcontext train-pcontext test-pcontext)
-    (compute-pcontexts pts+exs (*context*)))
+  (define fn (make-search-func (test-pre test) (list (test-input test)) (*context*)))
+  (for/list ([pt pts])
+    (define-values (result prec exs) (ival-eval fn pt))
+    (list pt (match exs
+               [(list (ival lo hi)) ((representation-bf->repr repr) lo)]
+               [(? nan?) (real->repr +nan.0 repr)]))))
```

Comment:
I think this can call eval-prog-real?

Reply:
Also who calls `get-exacts`?

Reply:
No one. 3 of the backends are unused. There's a test or two for each, but these are in danger of bit-rotting for sure.

Reply:
Yeah I'm OK keeping them but we need better coordination with Odyssey for sure.